.github/workflows/flowzone.yml

```diff
@@ -387,6 +389,111 @@ jobs:
             echo "enabled=false" >> $GITHUB_OUTPUT
           fi
 
+      - name: Check for Python Poetry pyproject.toml
+        id: python_poetry
+        run: |
+          if test -f "pyproject.toml"
+          then
+            echo "found pyproject.toml"
+            if grep 'build-backend.*poetry' pyproject.toml
+            then 
+              echo "Poetry used"
+              echo "enabled=true" >> $GITHUB_OUTPUT
+              echo "PYTHON_VERSIONS=[]" >> $GITHUB_ENV
+            else
+              echo "Poetry not used"
+              echo "enabled=false" >> $GITHUB_OUTPUT
+            fi
+          else
+            echo "enabled=false" >> $GITHUB_OUTPUT
+          fi
+
+      # Check which Python 3.7+ versions meet the Poetry project requirements
+      - name: Install Poetry
+        if: steps.python_poetry.outputs.enabled == 'true'
+        run: |
+          pipx install poetry
+
+      - name: Set up Python 3.7
+        if: steps.python_poetry.outputs.enabled == 'true'
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.7"
+
+      - name: Check compatibility
+        if: steps.python_poetry.outputs.enabled == 'true'
+        run: |
+          error_check=`(poetry install --dry-run --only-root 2>&1 || true)`
+          if ! grep -q "Please change python\|Poetry was unable to find a compatible version" <<< $error_check
+          then
+            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.7"]')" >> $GITHUB_ENV
+          else
+            echo "Python 3.7 does not meet project requirements."
+          fi
```

Comment:
This bit works here but fails for the [bottom-up](https://github.com/company-os/bottom-up-model-python/actions/runs/3282958824/jobs/5407059428#step:23:31) project. I need to investigate.

Reply:
Okay, so poetry was not changing the `python env` accordingly to the Python setup during the checks. 
Apparently, `poetry env use` also triggers a check whether python meets the project requirements which is a lot quicker as other dependencies do not need to be resolved. Hence using `poetry env use` instead of `poetry install` to deduce if the current python version is fine saves a lot of time, i.e. for the bottom-up project the compatibility check went from ~2mins to 9 secs.